tests/manage/z_cluster/test_hugepages.py

```diff
@@ -0,0 +1,99 @@
+import logging
+import pytest
+
+from ocs_ci.helpers.sanity_helpers import Sanity
+from ocs_ci.ocs import constants
+from ocs_ci.utility.utils import (
+    enable_huge_pages,
+    disable_huge_pages,
+)
+from ocs_ci.ocs.node import (
+    wait_for_nodes_status,
+    get_nodes,
+)
+from ocs_ci.ocs.resources.pod import (
+    wait_for_pods_to_be_running,
+)
+from ocs_ci.framework.testlib import (
+    bugzilla,
+    skipif_ocs_version,
+    skipif_external_mode,
+    ignore_leftovers,
+    E2ETest,
+    tier2,
+)
+
+log = logging.getLogger(__name__)
+
+
+@tier2
+@skipif_external_mode
+@skipif_ocs_version("<4.8")
+@bugzilla("1995271")
+@bugzilla("1995271")
+@pytest.mark.polarion_id("OCS-2754")
+@ignore_leftovers
+class TestHugePages(E2ETest):
+    """
+    Enable huge pages post cluster deployment
+
+    """
+
+    @pytest.fixture(scope="function", autouse=True)
+    def huge_pages_setup(self, request):
+        """
+        Initializes sanity
+
+        """
+        self.sanity_helpers = Sanity()
+
+        def finalizer():
+            """
+            Removes huge pages on worker nodes and verifies all pods are up
+
+            """
+            disable_huge_pages()
+
+            wait_for_nodes_status(status=constants.NODE_READY, timeout=600)
+
+            log.info("Wait for all pods to be in running state")
+            wait_for_pods_to_be_running(timeout=600)
+
```

Comment:
In a failure case, any chances that the node can go to not ready state while enabling huge pages? 
If yes, I would suggest you to add node_restart_teardown fixture

Reply:
Not in my testing, it looks stable. But yeah there are chances of failures like n/w etc. Added this teardown